### DISCUSSION TITLE
Can't add collection or field, but update content

### DISCUSSION THREAD

**Discussion by lmarschall** [pocketbase/pocketbase#2092] [Q&A]
Recently I wanted to update my collection and stumbled across different errors.
Adding fields to a collection fails, adding a collection itself fails as well, but I can alter my data inside an already existing collection.
Using the Chrome Dev Tools, I receive the following error:
index.abf6fd0a.js:4          PATCH https://pocketbase.my-custom-domain.com/api/collections/... 400
The PATCH Method Requests from altering the data are successfull though.

Error Message creating new Collection: "Failed to create the collection."
Error Message adding new Field: Something went wrong while processing your request.

Some Debugging, this seems to be the code failing in the index.js file: 
`this.beforeSend && (c = Object.assign({}, this.beforeSend(m, c))),
                [2, fetch(m, c).then(function(k) {`

Looking into the logs, I get the following error message, although I'm just reordering the fields...
`{
  "errorMessage": "type: cannot be blank."
}`

The fetch from the Google Dev Tools:
`fetch("https://pocketbase.my-custom-domain.com/api/collections/haysi6b00fyckx7", {
  "headers": {
    "accept": "*/*",
    "accept-language": "en-US",
    "authorization": "authToken",
    "cache-control": "no-cache",
    "content-type": "application/json",
    "pragma": "no-cache",
    "sec-ch-ua": "\"Google Chrome\";v=\"111\", \"Not(A:Brand\";v=\"8\", \"Chromium\";v=\"111\"",
    "sec-ch-ua-mobile": "?0",
    "sec-ch-ua-platform": "\"Linux\"",
    "sec-fetch-dest": "empty",
    "sec-fetch-mode": "cors",
    "sec-fetch-site": "same-origin"
  },
  "referrer": "https://pocketbase.my-custom-domain.com/_/",
  "referrerPolicy": "strict-origin-when-cross-origin",
  "body": "{\"id\":\"haysi6b00fyckx7\",\"created\":\"2022-10-26 10:46:29.661Z\",\"updated\":\"2022-10-26 10:52:49.611Z\",\"name\":\"products\",\"type\":\"\",\"system\":false,\"schema\":[{\"id\":\"jmft9tc4\",\"name\":\"name\",\"type\":\"text\",\"system\":false,\"required\":false,\"unique\":false,\"options\":{\"min\":null,\"max\":null,\"pattern\":\"\"},\"originalName\":\"name\",\"toDelete\":false},{\"id\":\"gocxfult\",\"name\":\"image\",\"type\":\"file\",\"system\":false,\"required\":false,\"unique\":false,\"options\":{\"maxSelect\":1,\"maxSize\":5242880,\"mimeTypes\":[],\"thumbs\":[]},\"originalName\":\"image\",\"toDelete\":false},{\"id\":\"bmekioyy\",\"name\":\"price\",\"type\":\"text\",\"system\":false,\"required\":false,\"unique\":false,\"options\":{\"min\":null,\"max\":null,\"pattern\":\"\"},\"originalName\":\"price\",\"toDelete\":false},{\"id\":\"wr2g1hfp\",\"name\":\"available\",\"type\":\"bool\",\"system\":false,\"required\":false,\"unique\":false,\"options\":{},\"originalName\":\"available\",\"toDelete\":false},{\"id\":\"gyff6cqj\",\"name\":\"unit\",\"type\":\"text\",\"system\":false,\"required\":false,\"unique\":false,\"options\":{\"min\":null,\"max\":null,\"pattern\":\"\"},\"originalName\":\"unit\",\"toDelete\":false},{\"id\":\"xnghfuz1\",\"name\":\"description\",\"type\":\"json\",\"system\":false,\"required\":false,\"unique\":false,\"options\":{},\"originalName\":\"description\",\"toDelete\":false},{\"id\":\"\",\"name\":\"field\",\"type\":\"text\",\"system\":false,\"required\":false,\"unique\":false,\"options\":{},\"originalName\":\"field\",\"toDelete\":false}],\"listRule\":\"\",\"viewRule\":\"\",\"createRule\":null,\"updateRule\":null,\"deleteRule\":null,\"options\":{},\"originalName\":\"products\"}",
  "method": "PATCH",
  "mode": "cors",
  "credentials": "include"
});`

I'm using version 0.10.0 behind a Traefik reverse proxy. Pocketbase itself is deployed via Kubernetes in a docker environment.
Anyone faced the same problems?

**Comment by ganigeorgiev** [comment]
Could you post the exact error message?

You can also find more detailed request errors in the "Admin UI > Logs" (make sure to toggle the "Include requests by admins").

_Side-note: If you have the time, I'd recommend updating to the latest v0.13.4 release. There were a lot of [changes](https://github.com/pocketbase/pocketbase/blob/master/CHANGELOG.md) since v0.10 and it is possible that the issue you are experiencing could be already fixed in a newer version._

**Reply by lmarschall** [reply]
Edited my initial post with some additional information. Updating the version is not really an option at the moment, because there is no compatibility after v0.10.0 and I would have to migrate the data manually :(

**Reply by ganigeorgiev** [reply]
I'm not sure how to reproduce it.

- Do you have any custom Go code that could interfere with the api request?

- Does the reverse proxy has any limitation on the body request size?

- Could you also share the submitted request data?
   You can find it in the browser's Inspect tool (for Chrome):
   Network tab > {find the create/update collection request} > Payload and Headers tab.


---

> because there is no compatibility after v0.10.0 and I would have to migrate the data manually

There are no major breaking changes to the public web APIs or the internal data structure after v0.10.
If you are using the prebuilt executable, the only change that eventually could require some manual migration steps would be probably [v0.11.0](https://github.com/pocketbase/pocketbase/blob/master/CHANGELOG.md#v0110) due to the introduced multi-match/match-all support:

> Note: Previously the behavior for multi-valued fields was as the "at least one of" type. The release comes with system db migration that will update your existing API rules (if needed) to preserve the compatibility. **If you have multi-select or multi-relation filter checks in your client-side code** and want to preserve the old behavior, you'll have to prefix with ? your operators.

If you decide to upgrade, always make sure to have a backup. You can try test replacing the binary locally to see if something is breaking.

**Reply by ganigeorgiev** [reply]
You can also start PocketBase with `--debug` and check the executed INSERT or UPDATE collection sql query in the console (or in your stdout logs file).

**Reply by lmarschall** [reply]
Ok, thank you very much, I'll take a look into this in the upcoming weeks. Added a fetch request.

**Reply by ganigeorgiev** [reply]
@lmarschall There are too many unknown details in your setup and I'm not sure how to help or reproduce it.

The submitted collection body doesn't have a type. Are you sure that you are using PocketBase v0.10.0 and you are not confusing it with the JS SDK version (the PocketBase version number could be found in the bottom-right corner in the Admin UI)?

Did you previously used PocketBase <0.8.0? If yes, did you follow the migration steps from https://github.com/pocketbase/pocketbase/blob/master/CHANGELOG.md#migrate-from-v07x?

**Reply by lmarschall** [reply]
Yes, I defined the version in my YAML file. I think the best solution would be recreating my setup again. Thank you for your help.

**Reply by lmarschall** [reply]
@ganigeorgiev Hey I just debugged my setup locally and realised I've been using version 0.7.6 locally and deployed the data folder on my k3s with version 0.10.0. I was able to recreate the error by changing my local version to 0.10.0 and using the data folder from 0.7.6 locally. I'm sorry for the inconvenience, I didn't realise I was using two different versions locally and online the whole time :(
